PowerEditor/src/NppBigSwitch.cpp

```diff
@@ -1869,6 +1869,10 @@ LRESULT Notepad_plus::process(HWND hwnd, UINT message, WPARAM wParam, LPARAM lPa
 
 			drawDocumentMapColoursFromStylerArray();
 
+			drawAutocompleteColoursFromStylerArray();
+			AutoCompletion::drawAutocomplete(_pEditView);
+			AutoCompletion::drawAutocomplete(_pNonEditView);
+
 			// Update default fg/bg colors in Parameters for both internal/plugins docking dialog

```

Comment:
OTOH, since the background of other panels go with Scintilla's background while theme's changing, I think it would be better that the auto-completion adapts the same behaviour, to have the set of colours on the fly. 

Please consider the following code:

```cpp
...
			drawTabbarColoursFromStylerArray();

			drawDocumentMapColoursFromStylerArray();

			// Update default fg/bg colors in Parameters for both internal/plugins docking dialog
			const Style * pStyle = NppParameters::getInstance().getGlobalStylers().findByID(STYLE_DEFAULT);
			if (pStyle)
			{
				NppParameters::getInstance().setCurrentDefaultFgColor(pStyle->_fgColor);
				NppParameters::getInstance().setCurrentDefaultBgColor(pStyle->_bgColor);

				int rbv = GetRValue(pStyle->_bgColor);
				int gbv = GetGValue(pStyle->_bgColor);
				int bbv = GetBValue(pStyle->_bgColor);

				int rfv = GetRValue(pStyle->_fgColor);
				int gfv = GetGValue(pStyle->_fgColor);
				int bfv = GetBValue(pStyle->_fgColor);
				
				COLORREF bgDarker = RGB(rbv-20 <= 0 ? 0 : rbv - 20, gbv-20 <= 0 ? 0 : gbv - 20, bbv-20 <= 0 ? 0 : bbv - 20);
				COLORREF fgDarker = RGB(rfv-20 <= 0 ? 0 : rfv - 20, gfv-20 <= 0 ? 0 : gfv - 20, bfv-20 <= 0 ? 0 : bfv - 20);
				COLORREF fgLigher = RGB(rfv+20 >= 255 ? 255 : rfv + 20, gfv + 20 >= 255 ? 255 : gfv + 20, bfv + 20 >= 255 ? 255 : bfv + 20);

				AutoCompletion::setColour(bgDarker, AutoCompletion::AutocompleteColorIndex::autocompleteBg);
				AutoCompletion::setColour(pStyle->_bgColor, AutoCompletion::AutocompleteColorIndex::selectedBg);
				AutoCompletion::setColour(fgDarker, AutoCompletion::AutocompleteColorIndex::autocompleteText);
				AutoCompletion::setColour(pStyle->_fgColor, AutoCompletion::AutocompleteColorIndex::selectedText);

				AutoCompletion::setColour(bgDarker, AutoCompletion::AutocompleteColorIndex::calltipBg);
				AutoCompletion::setColour(fgDarker, AutoCompletion::AutocompleteColorIndex::calltipText);
				AutoCompletion::setColour(fgLigher, AutoCompletion::AutocompleteColorIndex::calltipHighlight);
			}

			drawAutocompleteColoursFromStylerArray();
			AutoCompletion::drawAutocomplete(_pEditView);
			AutoCompletion::drawAutocomplete(_pNonEditView);

			NppDarkMode::calculateTreeViewStyle();
			auto refreshOnlyTreeView = static_cast<LPARAM>(TRUE);
...
```

In this way, we are not dependent on the DarkModeDefault.xml and theme xml files, and the auto-completion and calltip colours go with the most of themes (including DarkModeDefault.xml). 

What do you think?

Reply:
@donho 
good idea.
I've tried the above code and it works good for dark themes, but some light themes might have issue with highlight text.
Normal text and highlight text seem to have same color.